tests/lineage/test_lineage.py

```diff
@@ -118,6 +120,37 @@ def test_lineage_render(self, dag_maker):
         assert op1.inlets[0].url == f1s.format(DEFAULT_DATE)
         assert op1.outlets[0].url == f1s.format(DEFAULT_DATE)
 
+    def test_non_attr_outlet(self, dag_maker):
+        class A:
+            pass
+
+        a = A()
+
+        f3s = "/tmp/does_not_exist_3"
+        file3 = File(f3s)
+
+        with dag_maker(dag_id='test_prepare_lineage'):
+            op1 = EmptyOperator(
+                task_id='leave1',
+                outlets=[a, file3],
+            )
+            op2 = EmptyOperator(task_id='leave2', inlets='auto')
+
+            op1 >> op2
+
+        dag_run = dag_maker.create_dagrun(run_type=DagRunType.SCHEDULED)
+
+        ctx1 = Context({"ti": TI(task=op1, run_id=dag_run.run_id), "ds": DEFAULT_DATE})
+        ctx2 = Context({"ti": TI(task=op2, run_id=dag_run.run_id), "ds": DEFAULT_DATE})
+
+        # prepare with manual inlets and outlets
+        op1.pre_execute(ctx1)
+        op1.post_execute(ctx1)
```

Comment:
Thinking out loud, isn't it in-consistent that non-attr objects are not excluded to be set as "outlets" but excluded for inlets?

Example the following will evaluate to True currently on L149:

```
assert op1.outlets == [a, file3]
```

Reply:
Yeah possibly - I didn't really change this behaviour so I didn't think about it

Reply:
@kaxil WDYT? Leave this "as it was" or fix it now?

Reply:
Would lean towards leave this "as it was" in this PR and fix it in a separate PR